### PR TITLE
1892 Simplified login continue as

### PR DESCRIPTION
**Pull request by Daniel-Echegaray**

### CODE REVIEW

File: Sources/AccountSDKIOSWeb/Client/Client.swift [hunk 1]
```diff
@@ -115,14 +115,15 @@ public class Client: CustomStringConvertible {
         return authState
     }
     
-    private func createWebAuthenticationSession(withMFA: MFAType? = nil,
-                                                loginHint: String?,
-                                                extraScopeValues: Set<String> = [],
-                                                completion: @escaping LoginResultHandler) -> ASWebAuthenticationSession {
+    func createWebAuthenticationSession(withMFA: MFAType? = nil,
+                                        loginHint: String?,
```
Comment (zamorastp) [inline Sources/AccountSDKIOSWeb/Client/Client.swift:119]: for consistency can we provide a default value for loginHint? Its already done for other method arguments

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift [hunk 2]
```diff
@@ -78,6 +78,29 @@ struct SimplifiedLoginUIFactory {
...
+                        if #available(iOS 13.0, *) {
+                            let context = ASWebAuthSessionContextProvider()
+                            session.presentationContextProvider = context //TODO: Perhaps should be passed in
+                            session.prefersEphemeralWebBrowserSession = true
+                        }
+                        
+                        session.start()
+                    }
+                case .failure(let error):
+                    // TODO: How should we fail gracefully here
```
Comment (zamorastp) [inline Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:98]: Pop-up with a meaningful error for user should be good. But what next? hide simplified login?

File: Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift [hunk 3]
```diff
@@ -232,7 +232,14 @@ public struct SimplifiedLoginViewControllerRepresentable: UIViewControllerRepres
         let clientName = (Bundle.applicationName() != nil) ? Bundle.applicationName()! : "not_configured"
         let userContext = UserContextFromTokenResponse(identifier: "Demo identifier", display_text: "Display.Text", client_name: "demo client name")
         let userProfileResponse = UserProfileResponse(uuid: "", userId: "", status: nil, email: "demo@email.com", emailVerified: nil, emails: nil, phoneNumber: nil, phoneNumberVerified: nil, phoneNumbers: nil, displayName: "display.name", name: Name(givenName: "firstName", familyName: "lastName", formatted: nil), addresses: nil, gender: nil, birthday: nil, accounts: nil, merchants: nil, published: nil, verified: nil, updated: nil, passwordChanged: nil, lastAuthenticated: nil, lastLoggedIn: nil, locale: nil, utcOffset: nil)
-        let s = SimplifiedLoginUIFactory.buildViewController(client: client, userContext: userContext, userProfileResponse: userProfileResponse, clientName: clientName, withMFA: .password, loginHint: nil, extraScopeValues: [], withSSO: true, completion: completion) as! UINavigationController
+
```
Comment (zamorastp) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift:235]: our demo code, so I dont look at that :P

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift [hunk 4]
```diff
@@ -78,6 +78,29 @@ struct SimplifiedLoginUIFactory {
...
+            assertionFetcher.fetchAssertion { result in
+                switch result {
+                case .success(let assertion):
+                    DispatchQueue.main.async {
+                        let session = client.createWebAuthenticationSession(withMFA: nil, loginHint: nil, assertion: assertion.assertion, extraScopeValues: [], completion: completion)
+                        viewModel.asWebAuthenticationSession = session
+                        
+                        if #available(iOS 13.0, *) {
+                            let context = ASWebAuthSessionContextProvider()
+                            session.presentationContextProvider = context //TODO: Perhaps should be passed in
```
Comment (zamorastp) [inline Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:91]: can we somehow test that it should be passed?

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:91]: Or this is the same situation as for regular login flow

**Review by zamorastp — APPROVED** [review]

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift (hunk 4 above)
Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:91]: The regular login flow >13 passes the uiwindow in as a ContextProvider.
<13 does not rely on a passed in window/contextProvider.

I will create a new ticket for this. Having a uiwindow passed in could also alleviate, if we want the SDK to own the **presenting** of the SL-view,

**Pull request merged**